src/main/java/org/czekalski/petapiintegration/resourceServer/apiResource/v1/mapper/AnimalToAnimalResourceDto.java

```diff
@@ -0,0 +1,18 @@
+package org.czekalski.petapiintegration.resourceServer.apiResource.v1.mapper;
+
+import org.czekalski.petapiintegration.apiClient.v1.mapper.Mapper;
+import org.czekalski.petapiintegration.resourceServer.apiResource.v1.dto.AnimalResourceDto;
+import org.czekalski.petapiintegration.resourceServer.model.Animal;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AnimalToAnimalResourceDto implements Mapper<Animal, AnimalResourceDto> {
+    @Override
+    public AnimalResourceDto map(Animal animal) {
+        if (animal == null) {
```

Comment:
W Javie warto korzystać z `Optional`. W Kotlinie bezpieczeństwo zapewniają Ci typy, ale w Javie `Optional` to najelegantsze rozwiązanie takiego problemu. Może warto się zastanowić, czy `Mapper` nie powinien zwracać `Optional<T>`

Reply:
Nie jest to wprost powiedziane, ale chciałem, żebyś skorzystał `Pageable`. Z racji tego, że w odpowiedzi podana jest liczba stron i obecna strona, to tak niewprost jest powiedziane, że paginacja też powinna się znaleźć, a nie tylko limit

Reply:
Tzn.  miałem problem z tym aby wykorzystać Pageable bo to jest sprzężone z bazą danych z której Spring korzysta. Pageable należy do modułu Spring Data.

Reply:
Możesz wykorzystać ten moduł :)

Reply:
Nie wiem czy dobrze zrozumiałem.  W przypadku `Optional` to jeśli parametr ` animal` byłby nullem to wtedy miałoby to zwrócić `Optional.empty()` , a kiedy nie, to już normalny `Optional.of(...)`? Jakie są z tego benefity?

Reply:
https://www.youtube.com/watch?v=8idt_VpS6XQ&t=995s